### PR TITLE
Fix entry_points deprecation

### PR DESCRIPTION
**Pull request by felixdivo**
Part of  #1230.

### CODE REVIEW

**Review by zariiii9003 — APPROVED** [review]

File: can/interfaces/__init__.py [hunk 1]
```diff
@@ -32,14 +32,16 @@
...
+    except TypeError:
+        # Fallback for Python <3.10
+        # See https://docs.python.org/3/library/importlib.metadata.html#entry-points, "Compatibility Note"
+        entries = entry_points().get("can.interface", [])
+
+    BACKENDS.update(
+        {interface.name: tuple(interface.value.split(":")) for interface in entries}
+    )
 except ImportError:
     from pkg_resources import iter_entry_points
```
Comment (zariiii9003) [inline can/interfaces/__init__.py:46]: Do we still need this?

File: can/interfaces/__init__.py [hunk 2]
```diff
@@ -32,14 +32,16 @@
...
-            {
-                interface.name: tuple(interface.value.split(":"))
-                for interface in entry["can.interface"]
-            }
-        )
+    try:
+        entries = entry_points(group="can.interface")
+    except TypeError:
+        # Fallback for Python <3.10
+        # See https://docs.python.org/3/library/importlib.metadata.html#entry-points, "Compatibility Note"
```
Comment (zariiii9003) [inline can/interfaces/__init__.py:39]: We could also use `importlib_metadata` from PyPI as fallback

Comment (felixdivo) [reply can/interfaces/__init__.py:39]: That would also be possible, yep.
As it's just so little effort, I thought it's simpler to just handle the fallback here.

File: can/interfaces/__init__.py (hunk 1 above)
Comment (felixdivo) [reply can/interfaces/__init__.py:46]: Probably not for long, see #1234. 😃

**Pull request merged**